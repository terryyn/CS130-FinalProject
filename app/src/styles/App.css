html, body, #root{
	height: 100%;
	width: 100%;
}

:root {
	--mdc-theme-primary: #fcb8ab;
	--mdc-theme-secondary: #feeae6;
}

.App {
	height: 100%;
	width: 100%;
	display: flex;
}

.App-logo {
	height: 40vmin;
}

.App-header {
	background-color: #282c34;
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	font-size: calc(10px + 2vmin);
	color: white;
}

.App-link {
	color: #09d3ac;
}

#main-page {
	height: 100%;
	width: 100%;
	display: flex;
	flex-flow: row;
}
