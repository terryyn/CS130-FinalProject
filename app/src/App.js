import React, { useState, useEffect } from 'react';
import { BrowserRouter as Router, Switch, Route, Link, Redirect, useHistory, useLocation } from 'react-router-dom';


import Login from './containers/Login';
import Meeting from './containers/Meeting';
import Profile from './containers/Profile';
import Home from './containers/Home';
import Sidebar from './components/Sidebar';

import './styles/App.css';

import Server from './server';
const server = new Server();

function App() {
	const [ isLoggedIn, setLoggedIn ] = useState(false);
	const [ tryLogin, setTryLogin ] = useState(true);

	const [ currentUser, setUser ] = useState("");
	const [ currentUserPhotoUrl, setUserPhotoUrl ] = useState("https://cdn2.iconfinder.com/data/icons/animal-outline-icons-set/144/Dog-512.png");
	const [ currentUserEmail, setUserEmail ] = useState("");
	const [ currentIsInstructor, setcurrentIsInstructor ] = useState(false);

	useEffect(() => {
		if (tryLogin) {
			server.authenticateUser().then(data => {
				if (data!="fail") {
					setLoggedIn(true);
					setUserEmail(data.email);
					setUser(data.name);
					setcurrentIsInstructor(data.is_instructor);
				}
			})
		}
	});

	return (
		<div className="App">
			<Router>
				<Switch>
					<Route exact path="/">
						{isLoggedIn ?
							(
								<Redirect to='/home' />
							) :
							(<Login setLoggedIn={setLoggedIn} setTryLogin={setTryLogin} setUserPhotoUrl={setUserPhotoUrl} setUser={setUser} setUserEmail={setUserEmail} setcurrentIsInstructor={setcurrentIsInstructor}/>)
						}
					</Route>
					<Route path="/home">
						{isLoggedIn ?
							(
								<div id="main-page">
									<Sidebar setLoggedIn={setLoggedIn} setTryLogin={setTryLogin} currentUserPhotoUrl={currentUserPhotoUrl} currentPage={"home"}/>
									<Home currentUser={currentUser}/>
								</div>
							) :
							(<Redirect to='/' />)
						}
					</Route>
					<Route path="/profile">
						{isLoggedIn ?
							(
								<div id="main-page">
									<Sidebar setLoggedIn={setLoggedIn} setTryLogin={setTryLogin} currentUserPhotoUrl={currentUserPhotoUrl} currentPage={"profile"}/>
									<Profile currentUser={currentUser} currentUserPhotoUrl={currentUserPhotoUrl} currentUserEmail={currentUserEmail} currentIsInstructor={currentIsInstructor} setUser={setUser} setUserPhotoUrl={setUserPhotoUrl} setUserEmail={setUserEmail} setcurrentIsInstructor={setcurrentIsInstructor}/>
								</div>
							) :
							(<Redirect to='/' />)
						}
					</Route>
					<Route path="/meeting">
						{isLoggedIn ?
							(
								<div id="main-page">
									<Sidebar setLoggedIn={setLoggedIn} setTryLogin={setTryLogin} currentUserPhotoUrl={currentUserPhotoUrl} currentPage={"meeting"}/>
									<Meeting />
								</div>
							) :
							(<Redirect to='/' />)
						}
					</Route>
				</Switch>
			</Router>
		</div>
	);
}	

export default App;
