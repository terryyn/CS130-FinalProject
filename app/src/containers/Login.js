import React, { useState } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Divider from '@material-ui/core/Divider';

import { Link } from 'react-router-dom';
import FacebookLogin from 'react-facebook-login';
import '../styles/login.css';
import App from '../App'
import Server from '../server'

const exampleEmail = ["dinkarkhattar@gmail.com"];
const exampleUser = ["Dinkar Khattar"];

const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

function Login(props) {
  const [ newUser, setNewUser ] = useState(false);
  const [ name, setName ] = useState('');
  const [ email, setEmail ] = useState('');
  const [ password, setPassword ] = useState('');
  const [ isInstructor, setIsInstructor ] = useState(false);



  const classes = useStyles();

  let login = (response) => {
    console.log(response);
    if (exampleEmail.includes(response.email)) {
      props.setLoggedIn(true);
      props.setUser(response.name);
      props.setUserPhotoUrl(response.picture.data.url);
    }
    else {
      exampleUser.push(response.name);
      exampleEmail.push(response.email);
      props.setLoggedIn(true);
      props.setUser(response.name);
      props.setUserPhoto(response.picture.data.url);
    }
  }

  let regularLogin = (event) => {
    event.preventDefault();
    const form = {'username': name, 'email': email, 'password': password, 'isInstructor': isInstructor};
    const server = new Server();
    server.createUser(form);
  }

  let isNewUser = () => {
    if (newUser) {
      setNewUser(false);
    } else {
      setNewUser(true);
    }
	}

  return (
    <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper} id="main">
            <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
            Sign in
            </Typography>

            <div id="fbbutton">
              <FacebookLogin
              appId="1436766753144072"
              autoLoad
              fields="name,email,picture"
              callback={login}
              icon="fa-facebook"
              />
            </div>
            
            <hr />

            <div id="regLogin">
              <form className={classes.form} onSubmit={regularLogin} noValidate>
              {newUser &&
                  <TextField
                    autoComplete="name"
                    name="name"
                    variant="outlined"
                    required
                    fullWidth
                    id="name"
                    label="Name"
                    autoFocus
                    onInput={ e=>setName(e.target.value)}
                  />
              }
              <TextField
                  variant="outlined"
                  margin="normal"
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                  autoFocus
                  onInput={ e=>setEmail(e.target.value)}
              />
              <TextField
                  variant="outlined"
                  margin="normal"
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="current-password"
                  onInput={ e=>setPassword(e.target.value)}
              />
              <div class="new-user">
                {newUser &&
                  <FormControlLabel
                    control={<Checkbox checked={isInstructor} value="isInstructor" color="primary" />}
                    label="Are you an instructor?"
                    onChange={ e=>{
                        if (isInstructor) {
                          setIsInstructor(false)
                        }
                        else {
                          setIsInstructor(true)
                        }
                      }
                    }
                  />
                }
              </div>
              { !newUser &&
                <FormControlLabel
                    control={<Checkbox value="remember" color="primary" />}
                    label="Remember me"
                />
              }
              <Button
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
              >
                  Sign In
              </Button>
              <Grid container>
                  <Grid item xs>
                  <Link to="#">
                      Forgot password?
                  </Link>
                  </Grid>
                  <Grid item>
                  {!newUser ?
                    (
                      <Link onClick={isNewUser}>
                        {"Don't have an account? Sign Up"}
                      </Link>
                    ) :
                    (
                      <Link onClick={isNewUser}>
                        {"Have an account? Sign In"}
                      </Link>
                    )
                  }
                  </Grid>
              </Grid>
              </form>
            </div>
        </div>
    </Container>
  );
}

export default Login;