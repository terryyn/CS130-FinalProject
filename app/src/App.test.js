import React from 'react';
import App from './App.js';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import Login from './containers/Login.js';

const faker = require('faker');
const puppeteer = require('puppeteer');

describe('App', () =>{
  test('snapshot renders', () => {
    const component = renderer.create(<App/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('mock sign up', async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto('http://localhost:3000/');

    //make sure the log in page popos up
    await page.waitForSelector('#regLogin');

    // check nonexistent account
    // const signup = await page.$x("//Link[contains(text(),'Don\'t have an account? Sign Up')]");
    // if (signup) {
    //     await signup.click();
    // }
    await clickByText(page,"Don\'t have an account? Sign Up");

    await page.click('input[name=name');
    await page.type('input[name=name]','')
    await page.click('input[name=email]');
    await page.type('input[name=email]','yeyunong@hotmail.com');
    await page.click('input[name=password]');
    await page.type('input[name=password]','12345678');
    await page.keyboard.press('Enter');


    await page.$eval("#message-id", el => el.value = 'Invalid email. Must have a g.ucla.edu email');
  });
});

const clickByText = async (page, text) => {
  const escapedText = escapeXpathString(text);
  const linkHandlers = await page.$x(`//a[contains(text(), ${escapedText})]`);
  
  if (linkHandlers.length > 0) {
    await linkHandlers[0].click();
  } else {
    throw new Error(`Link not found: ${text}`);
  }
};

const escapeXpathString = str => {
  const splitedQuotes = str.replace(/'/g, `', "'", '`);
  return `concat('${splitedQuotes}', '')`;
};