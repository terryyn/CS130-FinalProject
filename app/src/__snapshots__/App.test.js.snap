// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      className="makeStyles-paper-1"
      id="main"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        id="text1"
      >
        Calendar of the Bruins
      </h4>
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        id="text2"
      >
        by the Bruins
      </h4>
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        id="text3"
      >
        for the Bruins
      </h4>
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0V0z"
            />
            <path
              d="M0 0h24v24H0V0z"
              opacity=".87"
            />
          </g>
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <hr />
      <div
        id="regLogin"
      >
        <form
          className="makeStyles-form-3"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            onInput={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Email Address
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="email"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-126 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-127"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            onInput={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
              <span
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="current-password"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-126 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-127"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="new-user"
          />
          <label
            className="MuiFormControlLabel-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-140 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-143"
                  data-indeterminate={false}
                  disabled={false}
                  onChange={[Function]}
                  type="checkbox"
                  value="remember"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Remember me
            </span>
          </label>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Sign in
            </span>
          </button>
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Forgot password?
              </a>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <a
                href=""
                onClick={[Function]}
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </main>
</div>
`;
